packages/framework/src/Model/Product/Product.php

```diff
@@ -893,6 +905,109 @@ protected function setTranslations(ProductData $productData)
         }
     }
 
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductData $productData
+     */
+    private function setDomains($productData)
+    {
+        $domainIds = array_keys($productData->seoTitles);
+        foreach ($domainIds as $domainId) {
+            try {
+                $productDomain = $this->getProductDomain($domainId);
+            } catch (ProductDomainNotFoundException $e) {
+                $productDomain = new ProductDomain($this, $domainId);
+                $this->domains[$domainId] = $productDomain;
+            }
+            $productDomain->setSeoTitle($productData->seoTitles[$domainId]);
+            $productDomain->setSeoH1($productData->seoH1s[$domainId]);
+            $productDomain->setSeoMetaDescription($productData->seoMetaDescriptions[$domainId]);
+            $productDomain->setDescription($productData->descriptions[$domainId]);
+            $productDomain->setShortDescription($productData->shortDescriptions[$domainId]);
+        }
+    }
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Model\Product\ProductDomain[]|\Doctrine\Common\Collections\ArrayCollection
+     */
+    public function getDomains()
+    {
+        return $this->domains;
+    }
+
+    /**
+     * @param int $domainId
+     * @return \Shopsys\FrameworkBundle\Model\Product\ProductDomain
+     */
+    public function getProductDomain(int $domainId)
+    {
+        foreach ($this->domains as $domain) {
+            if ($domain->getDomainId() === $domainId) {
+                return $domain;
+            }
+        }
+
+        throw new ProductDomainNotFoundException();
+    }
+
+    /**
+     * @param int $domainId
+     * @return string|null
+     */
+    public function getShortDescription(int $domainId)
+    {
+        return $this->getProductDomain($domainId)->getShortDescription();
+    }
+
+    /**
+     * @param int $domainId
+     * @return string|null
+     */
+    public function getDescription(int $domainId)
+    {
+        return $this->getProductDomain($domainId)->getDescription();
+    }
+
+    /**
+     * @param int $domainId
+     * @return string|null
+     */
+    public function getSeoH1(int $domainId)
+    {
+        return $this->getProductDomain($domainId)->getSeoH1();
+    }
+
+    /**
+     * @param int $domainId
+     * @return string|null
+     */
+    public function getSeoTitle(int $domainId)
+    {
+        return $this->getProductDomain($domainId)->getSeoTitle();
+    }
+
+    /**
+     * @param int $domainId
+     * @return string|null
+     */
+    public function getSeoMetaDescription($domainId)
+    {
+        return $this->getProductDomain($domainId)->getSeoMetaDescription();
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @return string
+     */
+    public function getSeoTitleForHtml(Domain $domain)
```

Comment:
Should this logic be in entity? In Brand, logic such as this was in the template.

Reply:
Okay, moved to template and refactored the old implementation of getting seo title in brand since it was wierdly conditioned and it was incosistent with other implementations.